build_tools/docker/samples/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# An image for running IREE's samples. Includes support for:
+#   * CMake
+#   * Vulkan (using SwiftShader)
+#   * Python (including `venv` and common pip packages needed for Colab)
+
+FROM gcr.io/iree-oss/cmake-python-swiftshader@sha256:75ad1156a486aa4bf3f93d45f9a7e0f052345fa103171dfc742f8c92d4a7ea0f AS final
+
+# Update setuptools per https://github.com/pypa/setuptools/issues/1694#issuecomment-466010982
+RUN apt-get update && apt-get install -y python3-venv python-setuptools && \
+    python3 -m pip install --upgrade setuptools
+
+# Note: we explicitly do *not* install Jupyter notebook requirements since
+# they should be installed within venvs.
+
+# Install additional packages often used in notebooks.
+# Installing these at the system level helps with caching, since venvs can
+# set --system-site-packages to use the already installed versions.
+RUN python3 -m pip install \
+    numpy \
+    matplotlib \
+    tensorflow \
```

Comment:
Elsewhere we use a pinned version of tf-nightly. I think that might be better here. Pinned versions for docker files in general are nice because it means that rebuilding the same docker image is more likely to give you the same result rather than someone changing something about swiftshader and then suddenly the samples tests are broken because this reran the install of tensorflow and now it's broken

Reply:
I agree with you in principle, but there are some details to consider here....

I tried basing off of https://github.com/google/iree/blob/98a6ea79b70e70d7022fbcda9464fd3ad8c99bff/build_tools/docker/cmake-bazel-frontends/Dockerfile#L22-L23
but decided against that after seeing various version skew warnings during pip installs (from TF).

For compatibility with Colab we could pin to whatever version Colab comes packaged with by default. https://colab.research.google.com/notebooks/tensorflow_version.ipynb indicates that is `2.5.0` right now.

Reply:
Pinning to the same version as colab SGTM. We might run into issues where we really need a later version though. I'm pretty sure there's a reason the other tests are at 2.6

Reply:
Pinned to what Colab uses today. We can rely on later versions outside of Colab, but until we have samples that would benefit from this sort of testing, I'm not going to reconcile those.